### ISSUE TITLE
Interrupting simplest example can hang process

### ISSUE THREAD

**Issue by cswarth** [BD2KGenomics/toil#1216]
Interrupting the [simplest quickstart example](http://toil.readthedocs.io/en/latest/running.html#running-a-workflow) can hang the process such that it must be killed it from another shell.

I don't have a repeatable test case other than running the program and hitting ^C at various times.

```
$ python --version
Python 2.7.12
$ pip freeze | grep -i toil
toil==3.3.3

$ python HelloWorld.py file:JobStore
stoat: 2016-10-18 11:17:10,703 INFO: toil.lib.bioio: Logging set at level: INFO
stoat: 2016-10-18 11:17:10,703 INFO: toil.lib.bioio: Logging set at level: INFO
stoat: 2016-10-18 11:17:10,709 INFO: toil.jobStores.fileJobStore: Path to job store directory is '/shared/silo_researcher/Matsen_F/MatsenGrp/working/cwarth/cft/JobStore'.
stoat: 2016-10-18 11:17:10,710 INFO: toil.jobStores.abstractJobStore: The workflow ID is: '86f48bbf-5ac0-4dd0-8833-19a0dddccc0e'
stoat: 2016-10-18 11:17:10,718 INFO: toil.common: Using the single machine batch system
stoat: 2016-10-18 11:17:10,718 WARNING: toil.batchSystems.singleMachine: Limiting maxCores to CPU count of system (16).
stoat: 2016-10-18 11:17:10,718 WARNING: toil.batchSystems.singleMachine: Limiting maxMemory to physically available memory (67378266112).
stoat: 2016-10-18 11:17:10,719 INFO: toil.common: Created the workflow directory at /tmp/toil-86f48bbf-5ac0-4dd0-8833-19a0dddccc0e
stoat: 2016-10-18 11:17:10,719 WARNING: toil.batchSystems.singleMachine: Limiting maxDisk to physically available disk (4970958848).
stoat: 2016-10-18 11:17:10,719 INFO: toil.batchSystems.singleMachine: Setting up the thread pool with 160 workers, given a minimum CPU fraction of 0.100000 and a maximum CPU value of 16.
stoat: 2016-10-18 11:17:10,781 INFO: toil.common: Written the environment for the jobs to the environment file
stoat: 2016-10-18 11:17:10,781 INFO: toil.common: Caching all jobs in job store
stoat: 2016-10-18 11:17:10,782 INFO: toil.common: 0 jobs downloaded.
stoat: 2016-10-18 11:17:10,795 INFO: toil.realtimeLogger: Real-time logging disabled
stoat: 2016-10-18 11:17:10,995 INFO: toil.leader: (Re)building internal scheduler state
stoat: 2016-10-18 11:17:10,996 INFO: toil.leader: Checked batch system has no running jobs and no updated jobs
stoat: 2016-10-18 11:17:10,996 INFO: toil.leader: Found 1 jobs to start and 0 jobs with successors to run
stoat: 2016-10-18 11:17:11,005 INFO: toil.leader: Starting the main loop
stoat: 2016-10-18 11:17:11,008 INFO: toil.batchSystems.singleMachine: Executing command: '_toil_worker file:JobStore H/S/jobujSXzy'.
stoat: 2016-10-18 11:17:11,476 INFO: toil.leader: No jobs left to run so exiting.
stoat: 2016-10-18 11:17:11,477 INFO: toil.leader: Finished the main loop
stoat: 2016-10-18 11:17:11,477 INFO: toil.leader: Waiting for stats and logging collator process to finish ...
stoat: 2016-10-18 11:17:11,536 INFO: toil.leader: ... finished collating stats and logs. Took 0.0587379932404 seconds
stoat: 2016-10-18 11:17:11,537 INFO: toil.leader: Waiting for service manager thread to finish ...
^C^CTraceback (most recent call last):
  File "HelloWorld.py", line 11, in <module>
    print Job.Runner.startToil(j, options) #Prints Hello, world!, ...
  File "/home/cwarth/.conda/envs/toil/lib/python2.7/site-packages/toil/job.py", line 500, in startToil
    return toil.start(job)
  File "/home/cwarth/.conda/envs/toil/lib/python2.7/site-packages/toil/common.py", line 564, in start
    self._shutdownBatchSystem()
  File "/home/cwarth/.conda/envs/toil/lib/python2.7/site-packages/toil/common.py", line 801, in _shutdownBatchSystem
    self._batchSystem.shutdown()
  File "/home/cwarth/.conda/envs/toil/lib/python2.7/site-packages/toil/batchSystems/singleMachine.py", line 241, in shutdown
    inputQueue.put(None)
  File "/home/cwarth/.conda/envs/toil/lib/python2.7/Queue.py", line 118, in put
    self.not_full.acquire()
KeyboardInterrupt
^C^C^C^C^C^C^C^C^C^C
```

No prompt came back and must be killed by sending `kill -9`

**Comment by cket** [comment]
We tried to fix this via #1096 but googling around led me to https://bugs.python.org/issue1360 which makes sense given your traceback

**Comment by hannes-ucsc** [comment]
Why would `put()` block? Is the queue bounded?

**Comment by cket** [comment]
Just realized that this version, 3.3.3, did not include the fix in #1096 that I pointed out. ceae644a88be5d4f6f03ee953739f0f8577aab6a should have fixed this on master.
